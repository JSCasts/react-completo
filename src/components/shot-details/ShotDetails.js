import React, { Component } from 'react';
import { Link } from 'react-router';

import InsertHtml from '../insert-html/InsertHtml';
import parseDate from '../../services/date';
import { getShotById } from '../../services/shots';
import closeButton from '../../assets/images/close.svg';
import styles from './styles.scss';

class ShotDetails extends Component {
  constructor(props) {
    super(props);

    this.state = {
      shot: null,
    };
  }

  componentWillMount() {
    getShotById(this.props.params.id).then(resp => this.setState({ shot: resp.data }));
  }

  render() {
    const { shot } = this.state;

    if (shot) {
      return (
        <div className={styles.container}>
          <div className={styles.main}>
            <img src={shot.user.avatar_url} alt="User avatar" />
            <div className={styles.header}>
              <h2>{shot.title}</h2>
              <h3>By <a href={shot.user.html_url}>
                {shot.user.name}</a> on {parseDate(shot.created_at)}
              </h3>
            </div>
            <Link to={'/'}>
              <figure className={styles.close}>
                <img src={closeButton} alt="Close button" />
              </figure>
            </Link>
          </div>
          <div className={styles.content}>
            <img src={shot.images.normal} alt="Shot" />
            <InsertHtml className={styles.description} text={shot.description} />
          </div>
        </div>
      );
    }
    return null;
  }
}

ShotDetails.propTypes = {
  params: React.PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default ShotDetails;
