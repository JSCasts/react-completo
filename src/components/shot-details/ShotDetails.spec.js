import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ShotDetails from './ShotDetails';

const props = {
  params: {
    id: 1,
  },
};

test('Snapshot', () => {
  let component = renderer.create(
    <ShotDetails {...props} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('ShotDetails', () => {
  test('has state', () => {
    const component = shallow(
      <ShotDetails {...props} />
    );
    expect(component.state()).toEqual({ shot: null });
  });

  describe('when state is null', () => {
    test('has text', () => {
      const component = shallow(
        <ShotDetails {...props} />
      );
      expect(component.text()).toBe('');
    });
  });

  describe('when state is defined', () => {
    test('has text', () => {
      const component = shallow(
        <ShotDetails {...props} />
      );

      const shot = {
        title: 'title',
        description: 'description',
        user: {
          avatar_url: 'avatar_url',
          html_url: 'html_url',
          name: 'name',
        },
        created_at: '2010-01-01',
        images: {
          normal: 'normal'
        },
      };

      component.setState({ shot });
      
      expect(component.html()).toContain(shot.title);
      expect(component.html()).toContain(shot.description);
      expect(component.html()).toContain(shot.user.avatar_url);
      expect(component.html()).toContain(shot.user.html_url);
      expect(component.html()).toContain(shot.user.name);
      expect(component.html()).toContain(shot.images.normal);
      expect(component.html()).toContain(shot.created_at);
    });
  });
});
