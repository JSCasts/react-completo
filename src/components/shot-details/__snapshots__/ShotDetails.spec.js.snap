exports[`test Snapshot 1`] = `null`;
